jetbrains-core/src/software/aws/toolkits/jetbrains/components/telemetry/ClientTelemetryComponent.kt

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.components.telemetry
+
+import com.intellij.openapi.components.ApplicationComponent
+import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.services.toolkittelemetry.ToolkitTelemetryClient
+import software.aws.toolkits.core.telemetry.BatchingMetricsPublisher
+import software.aws.toolkits.core.telemetry.ClientTelemetryPublisher
+import software.aws.toolkits.core.telemetry.MetricUnit
+import software.aws.toolkits.jetbrains.AwsToolkit
+import software.aws.toolkits.jetbrains.settings.AwsSettings
+import java.net.URI
+import java.time.Duration
+import java.time.ZonedDateTime
+
+interface ClientTelemetryComponent : ApplicationComponent
+
+class DefaultClientTelemetryComponent : ClientTelemetryComponent {
+    private lateinit var publisher: BatchingMetricsPublisher
+    private lateinit var startupTime: ZonedDateTime
+
+    override fun initComponent() {
+        super.initComponent()
+
+        publisher = BatchingMetricsPublisher(ClientTelemetryPublisher(
+            "AWS Toolkit For JetBrains",
+            AwsToolkit.PLUGIN_VERSION,
+            AwsSettings.getInstance().clientId.toString(),
+            ToolkitTelemetryClient
+                    .builder()
+                    // TODO: This is the beta endpoint. Replace with the production endpoint before release.
+                    .endpointOverride(URI.create("https://7zftft3lj2.execute-api.us-east-1.amazonaws.com/Beta"))
+                    // TODO: Determine why this client is not picked up by default.
+                    .httpClient(ApacheHttpClient.builder().build())
```

Comment:
Use `AwsSdkClient.getInstance())`, it has checks in it to prevent calling AWS on wrong threads

Reply:
✔️